src/partitioning/block.jl

```diff
@@ -36,21 +35,18 @@ function partition(object, method::BlockPartition)
   start   = @. ce - nleft * psides
   nblocks = @. nleft + nright
 
-  subsets   = [Vector{Int}() for i in 1:prod(nblocks)]
-  neighbors = [Vector{Int}() for i in 1:prod(nblocks)]
+  subsets   = [Int[] for i in 1:prod(nblocks)]
+  neighbors = [Int[] for i in 1:prod(nblocks)]
 
   # Cartesian to linear indices
   linear = LinearIndices(Dims(nblocks))
 
-  coords = MVector{Dim,T}(undef)
   for j in 1:nelements(object)
-    coordinates!(coords, object, j)
+    coords = coordinates(centroid(object, j))
 
     # find block coordinates
     c = @. floor(Int, (coords - start) / psides) + 1
-    @inbounds for i in 1:Dim
-      c[i] = clamp(c[i], 1, nblocks[i])
-    end
+    c = @. clamp(c, 1, nblocks)
```

Comment:
Why use the `@.` macro here instead of broadcasting `clamp` manually?

Reply:
I think just for consistency with the above line, I could have used it directly in the clamp call indeed. Does it make a difference in terms of performance?